### PR TITLE
extract missing deps via set overlaps with providing pkgs

### PR DESCRIPTION
**Pull request by CJ-Wright**

### CODE REVIEW

File: conda_forge_tick/audit.py [hunk 1]
```diff
@@ -54,6 +56,8 @@
     "versioneer",
     # not a real dep
     "cross-python",
+    "c_compiler_stub",
+    "cxx_compiler_stub",
```
Comment (beckermr) [inline conda_forge_tick/audit.py:60]: you might as well put the rest of them in here, especially fortran_compiler_stub and cuda_compiler_stub

Comment (CJ-Wright) [reply conda_forge_tick/audit.py:60]: Do we have a list of all the stubs?

Comment (CJ-Wright) [reply conda_forge_tick/audit.py:60]: nevermind I got it

File: conda_forge_tick/audit.py [hunk 2]
```diff
@@ -50,14 +50,18 @@
...
+    {
+        "python",
+        "setuptools",
+        "pip",
+        "versioneer",
+        # not a real dep
+        "cross-python",
+    }
+    | BUILTINS
+    | set(COMPILER_STUBS_WITH_STRONG_EXPORTS)
```
Comment (beckermr) [inline conda_forge_tick/audit.py:63]: This is fine, but do be aware that this is not the full list of possible stubs. Any compiler that only statically links is not in this list. I think this code is fine. This comment is an FYI.

Comment (CJ-Wright) [reply conda_forge_tick/audit.py:63]: Sure, and we should document all the stubs somewhere, but something tells me that the actual answer is hidden somewhere in conda-build or pinnings, and I didn't have my spelunking hat on :construction_worker_man:

Comment (beckermr) [reply conda_forge_tick/audit.py:63]: Yep. The answer is in pinnings and changes as that file changes. So much fun!

**Comment by beckermr** [comment]
huzzah!

**Pull request merged**